data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/TemporalTypeHandler.java

```diff
@@ -4,22 +4,133 @@
-        // Date and Time types
-        switch (columnType) {
-            // TODO: Implement the transformation
-            case DATE:
-            case TIME:
-            case TIMESTAMP:
-            case DATETIME:
-            case YEAR:
-                return value.toString();
-            default:
-                throw new IllegalArgumentException("Unsupported temporal data type: " + columnType);
+    public Long handle(final MySQLDataType columnType, final String columnName, final Object value,
+                       final TableMetadata metadata) {
+        if (value == null) {
+            return null;
+        }
+
+        final String strValue = value.toString().trim();
+
+        try {
+            switch (columnType) {
+                case DATE:
+                    return handleDate(strValue);
+                case TIME:
+                    return handleTime(strValue);
+                case DATETIME:
+                case TIMESTAMP:
+                    return handleDateTime(strValue);
+                case YEAR:
+                    return handleYear(strValue);
+                default:
+                    throw new IllegalArgumentException("Unsupported temporal data type: " + columnType);
+            }
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    String.format("Failed to parse %s value: %s", columnType, strValue), e);
+        }
+    }
+
+    private Long handleTime(final String timeStr) {
+        try {
+            // Try parsing as Unix timestamp first
+            final Long timeEpoch = parseDateTimeStrAsEpoch(timeStr);
+            if (timeEpoch != null) return timeEpoch;
+
+            final LocalTime time = LocalTime.parse(timeStr, TIME_FORMATTER);
+            // Combine with date from EPOCH
+            return time.atDate(LocalDate.EPOCH)
+                    .toInstant(ZoneOffset.UTC)
+                    .toEpochMilli();
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid time format: " + timeStr, e);
+        }
+    }
+
+    private Long handleDate(final String dateStr) {
+        try {
+            // Try parsing as Unix timestamp first
+            final Long dateEpoch = parseDateTimeStrAsEpoch(dateStr);
+            if (dateEpoch != null) return dateEpoch;
+
+            LocalDate date = LocalDate.parse(dateStr, DATE_FORMATTER);
+            return date.atStartOfDay(ZoneId.systemDefault())
+                    .withZoneSameInstant(ZoneOffset.UTC)
+                    .toInstant()
+                    .toEpochMilli();
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format: " + dateStr, e);
+        }
+    }
+
+    private Long handleDateTime(final String dateTimeStr) {
+        try {
+            final Long dateTimeEpoch = parseDateTimeStrAsEpoch(dateTimeStr);
+            if (dateTimeEpoch != null) return dateTimeEpoch;
```

Comment:
Similarly here, is the return value epoch in millis or epoch in micros?

Reply:
It's epoch seconds